cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -542,14 +542,21 @@ class SimpleArrayMixinCalculators
         auto athis = static_cast<A *>(this);
         if constexpr (!std::is_same_v<bool, std::remove_const_t<value_type>>)
         {
-            for (size_t i = 0; i < athis->size(); ++i)
+            const value_type * const end = athis->end();
```

Comment:
I guess this is what you meant by the comment:

> I found that non-SIMD arithmetic can be twice faster when implemented using raw pointer operation instead of the original operation that always get and set the value using `SimpleArray` public interface.
> 
> The non-SIMD operation therefore is improved using raw pointer operation.

It is interesting.  I would expect `SimpleArray::data()` to have zero runtime overhead.

A follow-up task is to study the assembly between the arithmetic operations using `data()` and raw pointers to understand why there is the runtime difference.  The knowledge will be useful for improving `SimpleArray` element-wise API.